logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/NewConversationEventHandler.kt

```diff
@@ -15,13 +21,28 @@ interface NewConversationEventHandler {
 }
 
 internal class NewConversationEventHandlerImpl(
-    private val conversationRepository: ConversationRepository
+    private val conversationRepository: ConversationRepository,
+    private val userRepository: UserRepository,
+    private val selfTeamIdProvider: SelfTeamIdProvider,
+    private val idMapper: IdMapper = MapperProvider.idMapper(),
+    private val memberMapper: MemberMapper = MapperProvider.memberMapper(),
 ) : NewConversationEventHandler {
     private val logger by lazy { kaliumLogger.withFeatureId(KaliumLogger.Companion.ApplicationFlow.EVENT_RECEIVER) }
 
-    override suspend fun handle(event: Event.Conversation.NewConversation) =
-        conversationRepository.insertConversationFromEvent(event).flatMap {
-            conversationRepository.updateConversationModifiedDate(event.conversationId, Clock.System.now().toString())
-        }.onFailure { logger.e("failure on new conversation event: $it") }
+    override suspend fun handle(event: Event.Conversation.NewConversation): Either<CoreFailure, Unit> = conversationRepository
+        .persistConversations(listOf(event.conversation), selfTeamIdProvider().getOrNull()?.value, originatedFromEvent = true)
+        .flatMap { conversationRepository.updateConversationModifiedDate(event.conversationId, Clock.System.now().toString()) }
+        .flatMap {
+            val members = memberMapper.fromApiModel(event.conversation.members)
+            conversationRepository.persistMembers(
+                listOf(members.self)
+                    .plus(members.otherMembers), event.conversationId
+            )
+        }
```

Comment:
Why insert members here? It's already done by the `persistConversations` function.

Reply:
I missed that, removed